src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -439,12 +440,16 @@ function loadOperation(
             loadBodyParameter(program, cadlParameters.bodyParameter)
         );
     } else if (cadlParameters.bodyType) {
-        const effectiveBodyType = getEffectiveSchemaType(
-            program,
-            cadlParameters.bodyType
-        );
-        if (effectiveBodyType.kind === "Model") {
-            parameters.push(loadBodyParameter(program, effectiveBodyType));
+        if (resourceOperation) {
```

Comment:
@chunyu3 , can you please add a reference what is a resource operation?

Reply:
https://github.com/Azure/cadl-azure/tree/main/packages/cadl-azure-core#define-resource-operations

resourceOperation is defined for azure resource, it will use a resource decorator '@createResouce', '@listResource', '@readResource', '@deleteResource' and so on.

cadl rest will provide an API to get the resource operation information if it is resource operation. https://github.com/microsoft/cadl/blob/c9a8ae555f8d36e30dc704b84dfbcc315215742c/packages/rest/src/rest.ts#L420